barcodes_generator_abstract/models/barcode_generate_mixin.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (C) 2014-TODAY GRAP (http://www.grap.coop)
+# Copyright (C) 2016-TODAY La Louve (http://www.lalouve.net)
+# Copyright 2017 LasLabs Inc.
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from odoo import _, api, exceptions, fields, models
+
+from .barcode_rule import _GENERATE_TYPE
+
+_logger = logging.getLogger(__name__)
+
+try:
+    import barcode
+except ImportError:
+    _logger.debug("Cannot import 'python-barcode' python library.")
+    barcode = None
+
+
+class BarcodeGenerateMixin(models.AbstractModel):
+    _name = "barcode.generate.mixin"
+    _description = "Generate Barcode Mixin"
+
+    # Column Section
+    barcode_rule_id = fields.Many2one(
+        string="Barcode Rule", comodel_name="barcode.rule"
+    )
+
+    barcode_base = fields.Integer(string="Barcode Base", copy=False)
+
+    generate_type = fields.Selection(
+        string="Generate Type",
+        selection=_GENERATE_TYPE,
+        readonly=True,
+        related="barcode_rule_id.generate_type",
+    )
+
+    @api.model
+    def create(self, vals):
+        """It creates a new barcode if automation is active."""
+        barcode_rule = self.env["barcode.rule"].get_automatic_rule(self._name)
+        if barcode_rule.exists():
+            vals.update({"barcode_rule_id": barcode_rule.id})
+        record = super(BarcodeGenerateMixin, self).create(vals)
+        if barcode_rule:
```

Comment:
```suggestion
        record = super().create(vals)
        if record.barcode_rule_id:
```

Reply:
The variable `barcode_rule` is defined on same method, I think what you suggest is not correct. https://github.com/OCA/stock-logistics-barcode/pull/317/files#diff-f50419781cc0e7e9172d09cf83bef00972791e23a60c5d8579585367c4453301R43 

I applied just the change on `super()`